server/server.go

```diff
@@ -240,6 +241,9 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 	handleFunc("/auth", s.handleAuthorization)
 	handleFunc("/auth/{connector}", s.handleConnectorLogin)
 	handleFunc("/callback", s.handleConnectorCallback)
+	// For easier connector-specific web server configuration, e.g. for the
+	// "external" connector.
+	handleFunc("/callback/{connector}", s.handleConnectorCallback)
```

Comment:
This needs to be the ID of the connector not the type, all connectors can be configured multiple times. I can configure two LDAP connectors with IDs `activedirectory` and `freeipa`.

It also needs enforce that `handleConnectorCallback` only allows a session using whatever value `{connector}` is. If it's `/callback/okta` that the user actually logging in through `okta`.

Reply:
> This needs to be the ID of the connector not the type, all connectors can be configured multiple times. I can configure two LDAP connectors with IDs activedirectory and freeipa.

Yes. The `/auth/{connector}` route assigns `connID := mux.Vars(r)["connector"]`, which is why I chose `/callback/{connector}` to signify the connector ID. Is that incorrect? Should “connector” be changed to “id”, here and above?

> It also needs enforce that handleConnectorCallback only allows a session using whatever value {connector} is. If it's /callback/okta that the user actually logging in through okta.

Done.